### ISSUE TITLE
no-unused-declaration false positives, not respecting disabling declarations

### ISSUE THREAD

**Issue by danvk** [cartant/eslint-plugin-etc#16]
In my `.eslintrc`:

```
    'etc/no-unused-declaration': [
      'error',
      {
        imports: true,
        declarations: false,
      }
    ]
```

In my code:

```
interface Foo {
  //      ~~~ "Foo" is not used; unused declarations are forbidden.eslintetc/no-unused-declaration
  x: string;
}

export interface Bar extends Foo {
  y: string;
}
```

First of all, this isn't an unused declaration since `Bar extends Foo`.
Second, why doesn't setting `declarations: false` disable this error?

Even these settings produce errors, which is surprising to me:

```
    'etc/no-unused-declaration': [
      'error',
      {
        imports: false,
        declarations: false,
      }
    ]
```

Version: 0.0.1-beta.41
